resharper/resharper-unity/src/CSharp/Daemon/Stages/PerformanceCriticalCodeAnalysis/PerformanceCriticalCodeAnalysisStage.cs

```diff
@@ -250,87 +293,78 @@ private static IMethod ExtractMethodDeclarationFromStartCoroutine([NotNull]ICSha
             {
                 var invocationReference = (coroutineInvocation.InvokedExpression as IReferenceExpression)?.Reference;
                 var info = invocationReference?.Resolve();
-                return info?.DeclaredElement as IMethod;
+                return info?.DeclaredElement;
             }
 
             return null;
         }
-
-        private void PropagateCostlyMark(IDeclaredElement node, IReadOnlyDictionary<IDeclaredElement, HashSet<IDeclaredElement>> callGraph,
-            HotMethodAnalyzerContext context, ISet<IDeclaredElement> visited)
-        {
-            if (visited.Contains(node))
-                return;
-            visited.Add(node);
-            context.MarkElementAsCostlyReachable(node);
-
-            if (callGraph.TryGetValue(node, out var children))
-            {
-                foreach (var child in children)
-                {
-                    PropagateCostlyMark(child, callGraph, context, visited);
-                }
-            }
-        }
-
-
+        
         // return value only for invocation nodes. If invoked method contain costly method `true` will be returned.
         private class HotMethodAnalyzer : TreeNodeVisitor<HotMethodAnalyzerContext>, IRecursiveElementProcessor<HotMethodAnalyzerContext>
         {
             private readonly IPsiSourceFile mySourceFile;
             private readonly IHighlightingConsumer myConsumer;
+            private readonly Func<bool> myCheckForInterrupt;
             private readonly int myMaxDepth;
 
-            public HotMethodAnalyzer(IPsiSourceFile sourceFile, IHighlightingConsumer consumer, int maxDepth)
+            public HotMethodAnalyzer(IPsiSourceFile sourceFile, IHighlightingConsumer consumer, Func<bool> checkForInterrupt, int maxDepth)
             {
                 mySourceFile = sourceFile;
                 myConsumer = consumer;
+                myCheckForInterrupt = checkForInterrupt;
                 myMaxDepth = maxDepth;
             }
 
+            public override void VisitMethodDeclaration(IMethodDeclaration methodDeclarationParam, HotMethodAnalyzerContext context)
+            {
+                context.MarkCurrentAsVisited();
```

Comment:
It looks like partial methods will have one random declaration processed (probably not even containing the actual code)